sdk/src/main/kotlin/io/rover/sdk/ui/blocks/poll/image/ImagePollOptionView.kt

```diff
@@ -0,0 +1,281 @@
+
+    private val bottomSection = relativeLayout {
+        gravity = Gravity.CENTER
+        setupLinearLayoutParams(
+            width = ViewGroup.LayoutParams.MATCH_PARENT,
+            height = bottomSectionHeight
+        )
+    }
+
+    private val bottomSectionLinear = linearLayout {
+        gravity = Gravity.CENTER
+        setupRelativeLayoutParams(
+            width = ViewGroup.LayoutParams.WRAP_CONTENT,
+            height = bottomSectionHeight
+        )
+    }
+
+    private val voteIndicatorView: AppCompatTextView = textView {
+        id = ViewCompat.generateViewId()
+        visibility = View.GONE
+        maxLines = 1
+        gravity = Gravity.CENTER
+
+        val marginInPixels = 8.dpAsPx(resources.displayMetrics)
+
+        setupLinearLayoutParams(
+            width = LayoutParams.WRAP_CONTENT,
+            height = bottomSectionHeight,
+            leftMargin = marginInPixels
+        )
+    }
+
+    private val optionImageView = imageView {}
+    private var roundRect: RoundRect? = null
+    private var borderPaint = Paint()
+    private var inResultsState = false
+
+    init {
+        setLayerType(View.LAYER_TYPE_SOFTWARE, null)
+        orientation = VERTICAL
+        topSection.addView(optionImageView)
+        topSection.addView(votePercentageView)
+        topSection.addView(votingIndicatorBar)
+        bottomSectionLinear.addView(optionTextView)
+        bottomSectionLinear.addView(voteIndicatorView)
+        bottomSection.addView(bottomSectionLinear)
+        addView(topSection)
+        addView(bottomSection)
+    }
+
+    fun bindOptionView(option: String, optionStyle: ImagePollBlockOptionStyle) {
+        val borderWidth = optionStyle.border.width.dpAsPx(this.resources.displayMetrics)
+        val horizontalMargin = 8f.dpAsPx(this.resources.displayMetrics)
+        bottomSection.layoutParams = (bottomSection.layoutParams as MarginLayoutParams).apply {
+            setMargins(borderWidth + horizontalMargin, 0, borderWidth + horizontalMargin, borderWidth * 2)
+        }
+
+        optionTextView.run {
+            text = option
+            textSize = optionStyle.font.size.toFloat()
+            setTextColor(optionStyle.color.asAndroidColor())
+            val font = optionStyle.font.weight.mapToFont()
+            typeface = Typeface.create(font.fontFamily, font.fontStyle)
+        }
+    }
+
+    fun bindOptionImageSize(imageLength: Int) {
+        topSection.layoutParams = LayoutParams(imageLength, imageLength)
+        optionImageView.run {
+            scaleType = ImageView.ScaleType.FIT_XY
+            layoutParams = RelativeLayout.LayoutParams(imageLength, imageLength)
+        }
+    }
+
+    fun bindOptionImage(bitmap: Bitmap) {
+        optionImageView.run {
+            setImageBitmap(bitmap)
+        }
+    }
+
+    fun goToResultsState(votingShare: Int, isSelectedOption: Boolean, optionStyle: ImagePollBlockOptionStyle) {
+        if (inResultsState) return
+        inResultsState = true
```

Comment:
What is the purpose of storing the state locally in the view? Just a performance optimization to avoid unnecessary view updates?

Reply:
I was trying to not let users keep swapping votes, but this isn't needed as I can just remove the click listeners of the option views when the `PollBlockViewModel` is in the results state. No need to store that state here

Reply:
awesome 👍